### PR TITLE
[image_picker] Removed cursor to prevent crash

### PR DESCRIPTION
**Pull request by AmolGangadhare**
Removed cursor and added MimeTypeMap.getFileExtensionFromUrl to prevent crash occurring in some scenarios.

## Description

*App is crashing in some scenarios while image picked from the file manager or other sources. In a line:182 while getting the mimeType for an image it was causing the app to crash as sometimes there is no mime type available for the image. Now, to get image extension cursor is removed and now extension will be taken from MimeTypeMap.getFileExtensionFromUrl*

## Related Issues

*Fixes flutter/flutter#35632*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/image_picker/android/build.gradle [hunk 1]
```diff
@@ -21,7 +21,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.3.0'
+        classpath 'com.android.tools.build:gradle:3.4.1'
```
Comment (cyanglaz) [inline packages/image_picker/android/build.gradle]: Seems not related to this PR?

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java [hunk 2]
```diff
@@ -168,25 +168,13 @@ private static String getPathFromRemoteUri(final Context context, final Uri uri)
...
-      if (cursor != null && cursor.moveToNext()) {
-        String mimeType = cursor.getString(0);
-
-        extension = MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType);
-      }
-    } finally {
-      if (cursor != null) {
-        cursor.close();
-      }
+      extension = MimeTypeMap.getFileExtensionFromUrl(uriImage.getPath());
```
Comment (cyanglaz) [inline packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java]: I did some research on this method and I found this: https://stackoverflow.com/questions/14320527/android-should-i-use-mimetypemap-getfileextensionfromurl-bugs/14321470
it seems getFileExtensionFromUrl is not sufficient for all scenarios?

File: packages/image_picker/android/build.gradle (hunk 1 above)
Comment (AmolGangadhare) [reply packages/image_picker/android/build.gradle]: gradle was updated after the PR made, it would be better if added with this PR else not necessary.

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java (hunk 2 above)
Comment (AmolGangadhare) [reply packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java]: agreed, would be better to find the extension manually but in some scenarios, I found that file uri doesn't have an extension so in that case, it will also not sufficient. so checking for null or empty extension and applying manual extension will be working.

Comment (cyanglaz) [reply packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java]: Sure, do you mind to add code to manually check the extension as well? 
Something like 
```
extension = fromURI
if (extension == null) {
  extension = manuallyParsing
}
if (extension == null) {
  extension = ".jpg" (i believe this is already handled)
}

```

File: packages/image_picker/android/build.gradle (hunk 1 above)
Comment (cyanglaz) [reply packages/image_picker/android/build.gradle]: ok, it'd be nice to rebase the PR so we won't see this diff here

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java (hunk 2 above)
Comment (AmolGangadhare) [reply packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java]: did the change, null check for extension was already there, in addition to that I added an empty check to prevent rare scenarios.

**Comment by cyanglaz** [comment]
LGTM! Please update the pubspec and changelog with the a new version, please also fix the formatting.

File: packages/image_picker/android/build.gradle (hunk 1 above)
Comment (cyanglaz) [reply packages/image_picker/android/build.gradle]: Sorry i misunderstood your comment here. I think what I wanted to say is that let's remove change in this particular PR, since it is not related to the PR. Also please remove the related bit in the CHANGELOG

Comment (AmolGangadhare) [reply packages/image_picker/android/build.gradle]: ok, I'll remove gradle change and comment from the changelog.

Comment (AmolGangadhare) [reply packages/image_picker/android/build.gradle]: @cyanglaz I've restored the gradle version and updated changelog.
